workstreams/Opaque_Pointers_Workstream.md

```diff
@@ -0,0 +1,98 @@
+# Workstream: Opaque Pointer Support
+
+## Motivation & Benefits
+
+Support for typed pointers in LLVM is being removed. More information about the
+motivation for this change can be found in LLVM forums. For example, the LLVM
+documentation on [opaque pointers](https://llvm.org/docs/OpaquePointers.html) as
+well as [this
+presentation](https://llvm.org/devmtg/2022-04-03/slides/keynote.Opaque.Pointers.Are.Coming.pdf)
+give a brief overview.
+
+The QIR specification currently makes use of typed pointers to represent qubit
+values and result values. While in LLVM 15, support for these types can still be
+enabled, an update to LLVM 16 requires us to change this representation.
```

Comment:
These versions are off by 1 possibly? There is no more typed pointers in version 17 and all future versions of LLVM, I believe. Since typed pointers were removed, there is no backwards compatibility by definition. Said another way, an LLVM front end user that requires typed pointers will be frozen at LLVM version 16.